diff --git a/source/inochi2d/core/format/serde/package.d b/source/inochi2d/core/format/serde/package.d
--- a/source/inochi2d/core/format/serde/package.d
+++ b/source/inochi2d/core/format/serde/package.d
@@ -3,6 +3,7 @@ import inochi2d.core.math;
 import inochi2d.core;
 import inmath.util;
 import std.traits;
+import nulib.string : isSomeNString;
 
 public import std.json : JSONValue, JSONType, parseJSON, toJSON;
 public import inochi2d.core.format.serde.deserializers;
@@ -101,6 +102,9 @@ void deserialize(T)(ref JSONValue data, ref T destination) {
 
     static if (is(T == JSONValue)) {
         destination = data;
+    } else static if (isSomeNString!T) {
+        if (!data.isNull)
+            destination = data.str;
     } else static if (isDeserializable!T) {
         static if (is(T == class) && is(typeof((ref T a) { a = new T; }))) {
             if (!destination)
