diff --git a/source/inochi2d/core/param/binding.d b/source/inochi2d/core/param/binding.d
--- a/source/inochi2d/core/param/binding.d
+++ b/source/inochi2d/core/param/binding.d
@@ -279,7 +279,7 @@ abstract class ParameterBindingImpl(T) : ParameterBinding {
     override
     void onDeserialize(ref JSONValue object) {
-        this.nodeRef = object.tryGetGUID("node", "target");
+        this.nodeRef = object.tryGetGUID("target", "node");
         object.tryGetRef(target.paramName, "param_name");
         object.tryGetRef(values, "values");
         object.tryGetRef(isSet_, "isSet");
