diff --git a/source/inochi2d/core/param/package.d b/source/inochi2d/core/param/package.d
--- a/source/inochi2d/core/param/package.d
+++ b/source/inochi2d/core/param/package.d
@@ -267,1 +267,2 @@
         object["guid"] = selfGuid.dup;
+        object["uuid"] = guid.toLegacyUUID();
diff --git a/source/inochi2d/core/param/binding.d b/source/inochi2d/core/param/binding.d
--- a/source/inochi2d/core/param/binding.d
+++ b/source/inochi2d/core/param/binding.d
@@ -267,1 +267,2 @@
-        object["node"] = nodeGuid.dup;
+        object["target"] = nodeGuid.dup;
+        object["node"] = target.node.guid.toLegacyUUID();
@@ -282,1 +283,1 @@
-        this.nodeRef = object.tryGetGUID("target", "node");
+        this.nodeRef = object.tryGetGUID("node", "target");
