diff --git a/source/inochi2d/core/nodes/drawable/part.d b/source/inochi2d/core/nodes/drawable/part.d
--- a/source/inochi2d/core/nodes/drawable/part.d
+++ b/source/inochi2d/core/nodes/drawable/part.d
@@ -58,6 +58,7 @@ class Part : Drawable {
         object["emissionStrength"] = emissionStrength;
         object["masks"] = masks.serialize();
+        object["mask_threshold"] = 0.5;
         object["opacity"] = opacity;
     }
 
