diff --git a/source/inochi2d/core/nodes/drivers/simplephysics.d b/source/inochi2d/core/nodes/drivers/simplephysics.d
--- a/source/inochi2d/core/nodes/drivers/simplephysics.d
+++ b/source/inochi2d/core/nodes/drivers/simplephysics.d
@@ -24,8 +24,8 @@ enum PhysicsModel {
     /**
         Rigid pendulum
     */
-    Pendulum = "pendulum",
+    Pendulum = "Pendulum",
 
     /**
         Springy pendulum
     */
-    SpringPendulum = "spring_pendulum",
+    SpringPendulum = "SpringPendulum",
 }
 
 enum ParamMapMode {
-    AngleLength = "angle_length",
-    XY = "xy",
-    LengthAngle = "length_angle",
-    YX = "yx",
+    AngleLength = "AngleLength",
+    XY = "XY",
+    LengthAngle = "LengthAngle",
+    YX = "YX",
 }
